carbonate_sync.py

```diff
@@ -0,0 +1,318 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import errno
+import getpass
+import itertools
+import logging
+import multiprocessing
+import os
+import progressbar
+import shutil
+import socket
+import subprocess
+import sys
+import tempfile
+import time
+import shlex
+import re
+
+from carbon import util
+import carbonate.cli as carbonate_cli
+import carbonate.config as carbonate_config
+import carbonate.cluster as carbonate_cluster
+import carbonate.sieve as carbonate_sieve
+import carbonate.sync as carbonate_sync
+import carbonate.util as carbonate_util
+
+TMP_DIR = (
```

Comment:
I would remove this global as it may be misleading (actual temp dir is from args) and it's just gettempdir()